IItemFoodEaten and GhostItemStack fix

kindlich opened the pull request.

kindlich committed:
Fix ghistItemStack setting damage isntead of amount

kindlich committed:
Added IItemFoodEaten for Food Item Representations

kindlich committed:
Added Braces to GhostItemStack#withTag method

SkySom commented on src/main/java/com/teamacronymcoders/contenttweaker/modules/vanilla/items/food/ItemContentFood.java:
```diff
@@ -68,30 +70,30 @@ public void setMod(IBaseMod mod) {
     public void checkFields() {
         List<String> missingFields = new ArrayList<>();
-        if (this.itemRepresentation.getUnlocalizedName() == null) {
+        if(this.itemRepresentation.getUnlocalizedName() == null) {
```
Looks like you're killing spaces. Double check your formatter :D

SkySom reviewed (changes requested):
Other than the killing of spaces in if statements, looks good, mind explaining the point of the new Food method?

kindlich committed:
Thou shalt add whitespaces

kindlich committed:
IItemUseFinish and EntityHelper

kindlich committed:
CreativeTab now defaults to 'items.png'

kindlich committed:
Merge remote-tracking branch 'origin/develop/1.12' into develop/1.12

kindlich committed:
Expanded CrT World to allow casting to CoT world.
Either use world as mods.contenttweaker.World (or the imported type)
or world.toContentTweakerWorld()
